tools/serve/serve.py

```diff
@@ -622,18 +622,18 @@ def __init__(self, mp_context, scheme=None):
     def start(self, init_func, host, port, paths, routes, bind_address, config, log_handlers, **kwargs):
         self.proc = self.mp_context.Process(target=self.create_daemon,
                                             args=(init_func, host, port, paths, routes, bind_address,
-                                                  config, log_handlers),
+                                                  config, log_handlers, dict(**os.environ)),
                                             name='%s on port %s' % (self.scheme, port),
                                             kwargs=kwargs)
         self.proc.daemon = True
         self.proc.start()
 
     def create_daemon(self, init_func, host, port, paths, routes, bind_address,
-                      config, log_handlers, **kwargs):
+                      config, log_handlers, env, **kwargs):
         # Ensure that when we start this in a new process we have the global lock
         # in the logging module unlocked
         importlib.reload(logging)
-
+        os.environ = env
```

Comment:
Is it possible we use `self.environ = env` instead?

Reply:
I don't really understand. This is running in the server subprocess. `os.environ` is the environment of that subprocess, but `self.environ` doesn't exist, and I don't see how we can achieve what we need without setting the process environment (to be honest I'm also not sure why we aren't inheriting the environment directly from the parent, but this does at least fix the issue).

Reply:
Thanks for the explanation. Looks I misunderstood previously.